api/v1beta1/machinedeployment_types.go

```diff
@@ -412,6 +417,31 @@ type RemediationStrategy struct {
 
 // ANCHOR_END: RemediationStrategy
 
+// MachineNamingStrategy allows changing the naming pattern used when creating
+// Machines.
+// Note: InfraMachines will use the same name as the corresponding Machines.
+type MachineNamingStrategy struct {
+	// Template defines the template to use for generating the names of the
```

Comment:
nit: replace "Template" with "template" so we use the serialized version of the field name https://github.com/kubernetes-sigs/cluster-api/issues/11238

Reply:
While we're at it, same for the KCP godoc. Thx!